docs/concepts/idempotence.md

```diff
@@ -0,0 +1,167 @@
+---
+id: idempotencia
+title: Idempotência
+tags:
+  - api
+  - conceito
+  - idempotencia
+---
+
+## Idempotência
+
+Idempotência, no contexto de desenvolvimento, significa que chamar uma determinada função uma vez ou várias vezes deve ter o mesmo efeito no estado do sistema.
+
+Uma função ser idempotente não significa que o retorno é sempre igual, mas que essa função, caso ela tenha alterado o estado do sistema, não faça nenhuma nova alteração, caso ela seja chamada logo em seguida com os mesmos parâmetros.
+
+Então, por exemplo, uma função que retorna o saldo de uma conta é idempotente, mesmo que ela retorne valores diferentes (porque outras funções podem ter alterado o estado do sistema entre uma chamada e outra.)
+
+### Por que idempotência é relevante?
+
+Um problema recorrente em sistemas orientados a eventos, é duplicação de eventos, isso pode ocorrer por diversas razões, como falhas de comunicação, invocações paralelas das funções que geram os eventos, etc.
+
+Sistemas resilientes são construídos considerando que esses problemas podem e vão frequentemente acontecer, e se preparando de maneira adequada para que falhas de comunicação não causem um estado interno ruim.
+
+Então, a principal importância da idempotência é fazer que seu sistema seja resiliente e confiável.
+
+### Quando uma função deve ser idempotente?
+
+Como tudo no desenvolvimento, depende fortemente do objetivo do projeto e das regras de negócio.
+
+Entretanto, existem alguns padrões:
+
+#### REST APIs
+
+##### Rotas PUT
```

Comment:
POST

Reply:
even though you can also do updates using POST, it is usually considered that PUT is the appropriate method for updates, which is what I meant here

maybe it needs further clarification